saleor/static/dashboard-next/pages/views/PageDetails.tsx

```diff
@@ -1,121 +1,139 @@
-                    }
-                  ) => {
-                    if (
-                      called &&
-                      !updateInProgress &&
-                      updateResult &&
-                      updateResult.pageUpdate &&
-                      !updateResult.pageUpdate.errors.length
-                    ) {
-                      if (error) {
-                        return;
-                      }
-                      return <Redirect to={pageListUrl} />;
-                    }
-                    return (
-                      <NavigatorLink to={pageListUrl}>
-                        {handleCancel => (
+                        return (
                           <>
-                            <WindowTitle title={maybe(() => data.page.title)} />
                             <PageDetailsPage
-                              onBack={handleCancel}
-                              page={loading ? undefined : data.page}
-                              onSubmit={data =>
-                                updatePage({
-                                  variables: { id, ...data }
+                              disabled={pageDetails.loading}
+                              errors={maybe(
+                                () => pageUpdateOpts.data.pageUpdate.errors,
+                                []
+                              )}
+                              saveButtonBarState={formTransitionState}
+                              page={pageDetails.data.page}
+                              onBack={() => navigate(pageListUrl)}
+                              onRemove={() => navigate(pageRemoveUrl(id))}
+                              onSubmit={formData =>
+                                pageUpdate({
+                                  variables: {
+                                    id,
+                                    input: {
+                                      content: JSON.stringify(formData.content),
+                                      isPublished: formData.isVisible
+                                        ? true
+                                        : formData.availableOn === "" ||
+                                          formData.availableOn === null
+                                        ? false
+                                        : true,
+                                      publicationDate: formData.isVisible
+                                        ? null
+                                        : formData.availableOn === ""
+                                        ? null
+                                        : formData.availableOn,
+                                      seo: {
+                                        description: formData.seoDescription,
+                                        title: formData.seoTitle
+                                      },
+                                      slug: formData.slug,
+                                      title: formData.title
```

Comment:
Personally I'd create some fn to compute those variables. The same goes for `formTransitionState` & `removeTransitionState` for cleaner jsx.